hardkernel/odroid-m1/default.nix

```diff
@@ -0,0 +1,33 @@
+# Based on kboot-conf by original authors povik and sstent.
+# I'm just porting things over. the original work can be viewed at:
+# https://github.com/sstent/nixos-on-odroid-m1 
+# https://github.com/povik/nixos-on-odroid-n2
+{ config, pkgs, lib, ... }:
+
+{
+  imports = [
+    ./kboot-conf
+  ];
+
+  #boot.loader.grub.enable = false;
+  boot.loader.kboot-conf.enable = true;
+  # Use kernel >6.6 The devicetree is missing from kernel versions older than this.
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "6.6") (lib.mkDefault pkgs.linuxPackages_latest);
+  # Stop ZFS breaking the build
+  boot.supportedFilesystems = lib.mkForce ["btrfs" "cifs" "f2fs" "jfs" "ntfs" "reiserfs" "vfat" "xfs"];
+
+  # I'm not completely sure if some of these could be omitted,
+  # but want to make sure disk access works
+  boot.initrd.availableKernelModules = [
```

Comment:
Can you run `nixos-generate-config` and see what modules it emits?

Reply:
Will try and get back to you.

Reply:
I ran `nixos-generate-config` and the generated list of available kernel modules remains empty. Here's the code I use to build my SD Image: https://github.com/KhashayarDanesh/nixos-on-odroid-m1